xt_training/utils/training.py

```diff
@@ -93,13 +95,18 @@ def train(args):
             if val_loader:
                 model.eval()
                 runner(val_loader, 'valid')
+                if use_nni:
+                    nni.report_intermediate_result(runner.loss().item())
 
             torch.save(model.state_dict(), f'{save_dir}/latest.pt')
             if runner.loss() < best_loss:
                 shutil.copy(f'{save_dir}/latest.pt', f'{save_dir}/best.pt')
                 best_loss = runner.loss()
                 print(f'Saved new best: {best_loss:.4}')
 
+        if use_nni:
+            nni.report_final_result(best_loss.item())
+    
```

Comment:
We should figure out what the early stopping does to the code here. If it acts like a keyboard interrupt (or some other exception type that we can handle), we should move the `nni.report_final_result(best_loss.item())` to after the except clause(s).